nixos/modules/services/continuous-integration/gitlab-runner.nix

```diff
@@ -116,10 +116,14 @@ in
         ++ optional hasDocker "docker.service";
       requires = optional hasDocker "docker.service";
       wantedBy = [ "multi-user.target" ];
+      preStart = "cp ${configFile} /var/lib/gitlab-runner/config.toml";
+      reload = "cp ${configFile} /var/lib/gitlab-runner/config.toml";
+      reloadIfChanged = true;
       serviceConfig = {
+        StateDirectory = "gitlab-runner";
         ExecStart = ''${cfg.package.bin}/bin/gitlab-runner run \
           --working-directory ${cfg.workDir} \
```

Comment:
what's done inside workingdirectory? does it make sense to point this to `StateDirectory` in default case? What about RuntimeDirectory?
Do we want to allow it to be overridden at all?

Reply:
I think this is mostly relevant if used witth the `shell` executor.